extensions/HyperVExtension/src/HyperVExtension/Models/VMGalleryCreationAdaptiveCardSession.cs

```diff
@@ -340,4 +335,22 @@ private async Task<ProviderOperationResult> HandleActionWhenFormInInitialState(A
 
         return (operationResult, shouldEndSession);
     }
+
+    private VMGalleryCreationUserInput GetAndValidateInput(string inputJson)
+    {
+        var deserializedObject = JsonSerializer.Deserialize(inputJson, typeof(VMGalleryCreationUserInput));
+        var inputForGalleryOperation = deserializedObject as VMGalleryCreationUserInput
+            ?? throw new InvalidDataException($"Json deserialization failed for input Json: {inputJson}");
+
+        var galleryImage = _vMGalleryImageList.Images[inputForGalleryOperation.SelectedImageListIndex];
+        var imageValidator = new VMImageSelectionValidator(galleryImage, _stringResource, _virtualMachineHost.VirtualHardDiskPath);
```

Comment:
Why not a static class/methods?

Reply:
No reason, I just preferred to make it a class. Since this is only used when the use clicks the next button I think we're fine without making it a static class.